includes/class-newspack-popups-inserter.php

```diff
@@ -62,13 +69,46 @@ public static function popups_for_post() {
 
 		// Retrieve all prompts eligible for display.
 		$popups_to_maybe_display = Newspack_Popups_Model::retrieve_eligible_popups( $include_unpublished, $campaign_id );
-
-		return array_filter(
+		$popups_to_display       = array_filter(
 			$popups_to_maybe_display,
 			function( $popup ) {
 				return self::should_display( $popup, true );
 			}
 		);
+
+		// Gather segments for all prompts to be displayed.
+		foreach ( $popups_to_display as $popup ) {
+			$assigned_segments = $popup['options']['selected_segment_id'];
+			if ( ! empty( $assigned_segments ) ) {
+				$assigned_segments = explode( ',', $assigned_segments );
+				foreach ( $assigned_segments as $segment_id ) {
+					if ( ! isset( self::$segments[ $segment_id ] ) ) {
+						$segment = Newspack_Segments_Model::get_segment( $segment_id );
+						if ( ! empty( $segment ) && ! empty( $segment['criteria'] ) ) {
+							self::$segments[ $segment_id ] = [
+								'criteria' => $segment['criteria'],
+								'priority' => $segment['priority'],
+							];
+						}
+					}
+				}
+			}
+		}
+
+		if ( ! empty( self::$segments ) ) {
+			add_action(
+				'wp_footer',
+				function() {
+					?>
+					<script>
+						var newspackPopupsSegments = <?php echo \wp_json_encode( self::$segments ); ?>;
+					</script>
+					<?php
+				}
+			);
+		}
```

Comment:
Any reason to favor inline JS instead of `wp_localize_script()`?

Reply:
Not at all—when I started I wasn't sure if we needed something other than JSON config for this, but I don't think we will.

Reply:
Changed to a localized value in e79cba4.